frontend/packages/console-shared/src/hooks/usePodsWatcher.ts

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { useK8sWatchResources } from '@console/internal/components/utils/k8s-watch-hook';
+import { getPodsDataForResource, getResourcesToWatchForPods } from '../utils';
+import { PodRCData } from '../types';
+
+export const usePodsWatcher = (
+  resource: K8sResourceKind,
+  kind?: string,
+  namespace?: string,
+): { loaded: boolean; loadError: string; podData: PodRCData } => {
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<string>('');
+  const [podData, setPodData] = React.useState<PodRCData>();
+  const watchKind = kind || resource.kind;
+  const watchNS = namespace || resource.metadata.namespace;
+  const watchedResources = React.useMemo(() => getResourcesToWatchForPods(watchKind, watchNS), [
+    watchKind,
+    watchNS,
+  ]);
+
+  const resources = useK8sWatchResources(watchedResources);
+
+  React.useEffect(() => {
+    const errorKey = Object.keys(resources).find((key) => resources[key].loadError);
+    if (errorKey) {
+      setLoadError(resources[errorKey].loadError);
+      return;
+    }
+    setLoadError(null);
```

Comment:
The initial value is `""` and typed as string. If the effect does not find an error this sets the error from `""` to `null` which force a re-rendering. Can we use `null` or `""` as initial state and here as well?

Reply:
@jeff-phillips-18 Thanks for removing the initial value "". But the default value is now undefined, and setting null will still re-render the component unnecessary. Using the exact same time in both cases or wrapping setLoadError(null) in an if would help.